Tests/SonarScanner.MSBuild.Common.Test/ProcessRunnerTests.cs

```diff
@@ -300,6 +300,56 @@ public void ProcRunner_ArgumentQuotingForwardedByBatchScript()
             AssertExpectedLogContents(testDir, expected);
         }
 
+        [TestMethod]
+        [WorkItem(1706)] // https://github.com/SonarSource/sonar-scanner-msbuild/issues/1706
+        public void ProcRunner_ArgumentQuotingScanner()
+        {
+            // Checks arguments passed to a batch script which itself passes them on are correctly escaped
+            var testDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var batchName = TestUtils.WriteBatchFileForTest(TestContext,
+@"@echo off
+REM The sonar-scanner.bat uses %* to pass the argument to javac.exe
+echo %*
+REM Because of the escaping, the single arguments are somewhat broken on echo. A workaround is to add some new lines for some reason. 
+echo %1
+
+
+echo %2
+
+
+echo %3
+
+
+echo %4
+
+
+");
+            var logger = new TestLogger();
+            var runner = new ProcessRunner(logger);
+            var expected = new[]
+            {
+                @"-Dsonar.scanAllFiles=true",
+                @"-Dproject.settings=D:\DevLibTest\ClassLibraryTest.sonarqube\out\sonar-project.properties",
+                @"--from=ScannerMSBuild/5.13.1",
+                @"--debug"
```

Comment:
Taken from the community ticket logs:
`-Dsonar.scanAllFiles=true -Dproject.settings=D:\DevLibTest\ClassLibraryTest\.sonarqube\out\sonar-project.properties --from=ScannerMSBuild/5.13 --debug <sensitive data removed>`
`<sensitive data removed>` indicates that there might be something passed that caused the failure but isn't in the logs.